### PR TITLE
Improve demo

### PR DESCRIPTION
**Pull request by emmanuelbernard**
Not necessarily to apply but here is how the demo looks like now:
- PU pointing to H2 as well for the ordering system
- added the notion of trip allowing several hikes
- added populate logic via URL trigger
- upgraded to WF 8.2 and OGM 4.1 final

### CODE REVIEW

**Comment by emmanuelbernard** [comment]
CC @gunnarmorling

**Comment by gunnarmorling** [comment]
@emmanuelbernard Seems like we can merge the first four commits either way because they are just the general update and clean-up?

File: hibernate-ogm/hiking-demo/src/test/java/org/hibernate/ogm/hiking/OrderRepositoryIT.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+			.addAsWebInfResource( new File( WEBAPP_SRC + "WEB-INF/beans.xml" ) )
+			.addAsResource( new StringAsset(
+					"Dependencies: org.hibernate:ogm services, org.hibernate.ogm.mongodb services" ),
+					"META-INF/MANIFEST.MF"
+			);
+	}
+
+	@Test
+	public void shouldPersistOrder() {
+		Order order = orderRepository.createOrder();
```
Comment (DavideD) [inline hibernate-ogm/hiking-demo/src/test/java/org/hibernate/ogm/hiking/OrderRepositoryIT.java]: There are a few compilation error in this class

OrderRepository#createOrder() does not exist

File: hibernate-ogm/hiking-demo/src/test/java/org/hibernate/ogm/hiking/OrderRepositoryIT.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		assertNotNull( order.number );
+		assertNotNull( order.id );
+
+		//entityManager.flush();
+
+		order = orderRepository.getOrderByNumber( order.number );
+
+		assertNotNull( order );
+		assertTrue( order.hikeId != 0 );
```
Comment (DavideD) [inline hibernate-ogm/hiking-demo/src/test/java/org/hibernate/ogm/hiking/OrderRepositoryIT.java]: Order#hikeId does not exist

File: hibernate-ogm/hiking-demo/src/test/java/org/hibernate/ogm/hiking/OrderRepositoryIT.java (hunk 1 above)
Comment (emmanuelbernard) [inline hibernate-ogm/hiking-demo/src/test/java/org/hibernate/ogm/hiking/OrderRepositoryIT.java]: Well done Emmanuel!
I'm on it.

**Comment by emmanuelbernard** [comment]
@DavideD This is fixed witha  pushed commit. Sorry
@gunnarmorling yes

**Comment by emmanuelbernard** [comment]
Rebased as a new PR https://github.com/hibernate/hibernate-demos/pull/12

**Pull request closed without merging**